[dialog] TS compilation error with TypeScript 4.4.3

### Description

When trying to update [lit-vaadin-helpers](https://github.com/web-padawan/lit-vaadin-helpers) to TypeScript to 4.4.3 I got the following errors:

```
node_modules/@vaadin/vaadin-dialog/src/vaadin-dialog.d.ts:80:3 - error TS2416: Property 'ariaLabel' in type 'DialogElement' is not assignable to the same property in base type 'HTMLElement & DialogResizableMixin & DialogDraggableMixin & ElementMixin & DirMixin & ThemePropertyMixin'.
  Type 'string | null | undefined' is not assignable to type 'string'.
    Type 'undefined' is not assignable to type 'string'.

80   ariaLabel: string | null | undefined;
     ~~~~~~~~~

node_modules/@vaadin/vaadin-dialog/src/vaadin-dialog.d.ts:111:3 - error TS2416: Property 'addEventListener' in type 'DialogElement' is not assignable to the same property in base type 'HTMLElement & DialogResizableMixin & DialogDraggableMixin & ElementMixin & DirMixin & ThemePropertyMixin'.
  Type '<K extends "input" | "progress" | "error" | "pause" | "play" | "waiting" | "abort" | "ended" | "resize" | "toggle" | "copy" | "select" | "fullscreenchange" | "fullscreenerror" | "cut" | ... 84 more ... | "opened-changed">(type: K, listener: (this: DialogElement, ev: DialogEventMap[K]) => void, options?: boolean | .....' is not assignable to type '{ <K extends keyof HTMLElementEventMap>(type: K, listener: (this: HTMLElement, ev: HTMLElementEventMap[K]) => any, options?: boolean | AddEventListenerOptions | undefined): void; (type: string, listener: EventListenerOrEventListenerObject, options?: boolean | ... 1 more ... | undefined): void; }'.
    Types of parameters 'listener' and 'listener' are incompatible.
      The 'this' types of each signature are incompatible.
        Type 'DialogElement' is not assignable to type 'HTMLElement'.
          Types of property 'ariaLabel' are incompatible.
            Type 'string | null | undefined' is not assignable to type 'string'.
              Type 'undefined' is not assignable to type 'string'.

111   addEventListener<K extends keyof DialogEventMap>(
```

### Expected outcome

There should be no errors.

### Actual outcome

Compilation fails.

### Proposed solution

We should update `ariaLabel` property with the one in `HTMLElement` to make it work.

--- a/package.json
+++ b/package.json
@@ -69,6 +69,6 @@
     "stylelint": "^13.13.1",
     "stylelint-config-prettier": "^8.0.2",
     "stylelint-config-vaadin": "^0.2.10",
-    "typescript": "4.3.5"
+    "typescript": "^4.4.3"
   }
 }

--- a/packages/app-layout/src/vaadin-drawer-toggle.d.ts
+++ b/packages/app-layout/src/vaadin-drawer-toggle.d.ts
@@ -10,7 +10,7 @@ import { Button } from '@vaadin/button/src/vaadin-button.js';
  * ```
  */
 declare class DrawerToggleElement extends Button {
-  ariaLabel: string | null | undefined;
+  ariaLabel: string;
 }
 
 declare global {

--- a/packages/crud/src/vaadin-crud-edit-column.d.ts
+++ b/packages/crud/src/vaadin-crud-edit-column.d.ts
@@ -19,7 +19,7 @@ declare class CrudEditColumnElement extends GridColumnElement {
   /**
    * The arial-label for the edit button
    */
-  ariaLabel: string | null | undefined;
+  ariaLabel: string;
 }
 
 declare global {

--- a/packages/vaadin-dialog/src/vaadin-dialog.d.ts
+++ b/packages/vaadin-dialog/src/vaadin-dialog.d.ts
@@ -73,9 +73,8 @@ declare class DialogElement extends ThemePropertyMixin(
 
   /**
    * Set the `aria-label` attribute for assistive technologies like
-   * screen readers. An `undefined` value for this property (the
-   * default) means that the `aria-label` attribute is not present at
-   * all.
+   * screen readers. An empty string value for this property (the
+   * default) means that the `aria-label` attribute is not present.
    */
   ariaLabel: string;
 

--- a/packages/vaadin-dialog/src/vaadin-dialog.d.ts
+++ b/packages/vaadin-dialog/src/vaadin-dialog.d.ts
@@ -77,7 +77,7 @@ declare class DialogElement extends ThemePropertyMixin(
    * default) means that the `aria-label` attribute is not present at
    * all.
    */
-  ariaLabel: string | null | undefined;
+  ariaLabel: string;
 
   /**
    * Custom function for rendering the content of the dialog.

--- a/packages/vaadin-dialog/src/vaadin-dialog.js
+++ b/packages/vaadin-dialog/src/vaadin-dialog.js
@@ -235,12 +235,12 @@ class DialogElement extends ThemePropertyMixin(
 
       /**
        * Set the `aria-label` attribute for assistive technologies like
-       * screen readers. An `undefined` value for this property (the
-       * default) means that the `aria-label` attribute is not present at
-       * all.
+       * screen readers. An empty string value for this property (the
+       * default) means that the `aria-label` attribute is not present.
        */
       ariaLabel: {
-        type: String
+        type: String,
+        value: ''
       },
 
       /**
@@ -317,7 +317,7 @@ class DialogElement extends ThemePropertyMixin(
 
   /** @private */
   _ariaLabelChanged(ariaLabel) {
-    if (ariaLabel !== undefined && ariaLabel !== null) {
+    if (ariaLabel != null && ariaLabel != '') {
       this.$.overlay.setAttribute('aria-label', ariaLabel);
     } else {
       this.$.overlay.removeAttribute('aria-label');

--- a/packages/vaadin-dialog/src/vaadin-dialog.js
+++ b/packages/vaadin-dialog/src/vaadin-dialog.js
@@ -317,7 +317,7 @@ class DialogElement extends ThemePropertyMixin(
 
   /** @private */
   _ariaLabelChanged(ariaLabel) {
-    if (ariaLabel != null && ariaLabel != '') {
+    if (ariaLabel) {
       this.$.overlay.setAttribute('aria-label', ariaLabel);
     } else {
       this.$.overlay.removeAttribute('aria-label');

--- a/packages/vaadin-dialog/test/dialog.test.js
+++ b/packages/vaadin-dialog/test/dialog.test.js
@@ -74,10 +74,21 @@ describe('vaadin-dialog', () => {
         expect(overlay.getAttribute('aria-label')).to.be.null;
       });
 
-      it('overlay should get an updated `aria-label` attribute (if changed)', () => {
+      it('overlay should not have `aria-label` attribute if set to undefined', () => {
         dialog.ariaLabel = 'accessible';
-        expect(overlay.getAttribute('aria-label')).to.be.eql('accessible');
         dialog.ariaLabel = undefined;
+        expect(overlay.getAttribute('aria-label')).to.be.null;
+      });
+
+      it('overlay should not have `aria-label` attribute if set to null', () => {
+        dialog.ariaLabel = 'accessible';
+        dialog.ariaLabel = null;
+        expect(overlay.getAttribute('aria-label')).to.be.null;
+      });
+
+      it('overlay should not have `aria-label` attribute if set to empty string', () => {
+        dialog.ariaLabel = 'accessible';
+        dialog.ariaLabel = '';
         expect(overlay.getAttribute('aria-label')).to.be.null;
       });
 

